sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingInterpreter.scala

```diff
@@ -86,9 +86,11 @@ case class SqlScriptingInterpreter(session: SparkSession) {
           .map(varName => DropVariable(varName, ifExists = true))
           .map(new SingleStatementExec(_, Origin(), args, isInternal = true))
           .reverse
-        new CompoundBodyExec(
-          collection.map(st => transformTreeIntoExecutable(st, args)) ++ dropVariables,
-          label)
+        var statements =
+          collection.map(st => transformTreeIntoExecutable(st, args)) ++ dropVariables
+        if (statements.isEmpty) statements = Seq(new NoOpStatementExec)
```

Comment:
We can maybe write this in more scala-like style to avoid using `var`.

```
val statements = 
  collection.map(st => transformTreeIntoExecutable(st, args)) ++ dropVariables match {
    case Nil => Seq(new NoOpStatementExec)
    case s => s
  }
```

Reply:
done